core-bundle/src/EventListener/DataContainer/FormOverwriteListener.php

```diff
@@ -0,0 +1,104 @@
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\Controller;
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\DataContainer;
+use Contao\FormModel;
+use Contao\System;
+
+class FormOverwriteListener
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    public function __construct(ContaoFramework $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    /**
+     * @Hook("loadDataContainer")
+     */
+    public function addOverwritableFields(string $table): void
+    {
+        if (!\in_array($table, ['tl_content', 'tl_module'], true)) {
+            return;
+        }
+
+        $this->framework->getAdapter(Controller::class)->loadDataContainer('tl_form');
+        $this->framework->getAdapter(System::class)->loadLanguageFile('tl_form');
+
+        $pm = PaletteManipulator::create();
+        $fields = $this->getOverwritableFields();
+
+        foreach ($fields as $field => $config) {
+            $targetField = 'form_'.$field;
+            $this->copyFieldConfig($table, $targetField, $config);
+            $pm->addField($targetField, 'form_legend');
+        }
+
+        $pm->applyToSubpalette('form_overwriteConfig', $table);
+    }
+
+    public function getPlaceholderFromForm($varValue, DataContainer $dc)
+    {
+        $formId = $dc->activeRecord->form;
+        $formModel = $this->framework->getAdapter(FormModel::class);
+
+        if (!$formId) {
+            return $varValue;
+        }
+
+        $form = $formModel->findById($formId);
+        $formField = str_replace('form_', '', $dc->field);
+
+        if (isset($form->{$formField})) {
+            $GLOBALS['TL_DCA'][$dc->table]['fields'][$dc->field]['eval']['placeholder'] = $form->{$formField};
+        }
+
+        return $varValue;
+    }
+
+    private function getOverwritableFields(): array
+    {
+        $overwritable = [];
+
+        foreach ($GLOBALS['TL_DCA']['tl_form']['fields'] as $field => $config) {
+            if (isset($config['eval'], $config['eval']['formOverwritable']) && true === $config['eval']['formOverwritable']) {
+                $overwritable[$field] = $config;
+            }
+        }
+
+        return $overwritable;
+    }
+
+    private function copyFieldConfig(string $table, string $targetField, array $dca): void
+    {
+        $GLOBALS['TL_DCA'][$table]['fields'][$targetField] = $dca;
+        $GLOBALS['TL_DCA'][$table]['fields'][$targetField]['eval']['mandatory'] = false;
+
+        if (!isset($GLOBALS['TL_DCA'][$table]['fields'][$targetField]['load_callback'])) {
+            $GLOBALS['TL_DCA'][$table]['fields'][$targetField]['load_callback'] = [];
+        }
+
+        $GLOBALS['TL_DCA'][$table]['fields'][$targetField]['load_callback'][] = [
+            self::class, 'getPlaceholderFromForm',
+        ];
+    }
+}
```

Comment:
Personally I would make the following adjustments:

* Use `__invoke` instead of `addOverwritableFields`.
* Put the annotation on the class, instead of the method.
* Rename the class to `AddOverwritableFormSettingsListener`.

Reply:
Yep, that sounds more easily comprehensible 👍

Reply:
After discussing it on Slack and trying out a couple things I cannot come up with a solution which is better. I added a comment to the actual callback implementation to clarify that it is registered dynamically and where this happens.